### PR TITLE
feat(launch): create launch agent includes agent config if present

### PR DESCRIPTION
**Pull request by TimH98**
Fixes
-----
- Fixes [WB-14421](https://wandb.atlassian.net/browse/WB-14421)

Description
-----------
The api function `create_launch_agent` can now take in an agent config and includes it in the call to the `createLaunchAgent` mutation.

### <samp>🤖 Generated by Copilot at f3bc1d0</samp>

This pull request adds a feature to create launch agents with dynamic mutation queries, fixes a bug in a launch test, and passes the default configuration options to the launch agent class. The feature improves the flexibility and efficiency of the launch agent creation process, the bug fix ensures the test matches the actual function signature, and the default configuration options simplify the agent initialization and API call.

Testing
-------
Tested in tandem with the [backend PR](https://github.com/wandb/core/pull/15965):

- Start gorilla: `governor start invoker.ini`
- `pip install -e {path to wandb repo}`
- Start an agent, pointing it to base url `http://localhost:9001`
- Identify the mysql docker container and exec into it: `docker exec -it {container ID} mysql`
- `use wandb_dev;`
- `select * from launch_agents;`
- Confirm that the newly started launch agent has the `agent_config` field set

### <samp>🤖 Generated by Copilot at f3bc1d0</samp>

> _We're the launch agents of the sea, we sail the code with glee_
> _We fix the bugs and add the features, we make the `Api` sing_
> _We heave and ho and pull the rope, we raise the yardarm high_
> _We use the schema and the config, we craft the query on the fly_

[WB-14421]: https://wandb.atlassian.net/browse/WB-14421?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -1528,12 +1529,38 @@ def ack_run_queue_item(self, item_id: str, run_id: Optional[str] = None) -> bool
         result: bool = response["ackRunQueueItem"]["success"]
         return result
 
+    @normalize_exceptions
+    def create_launch_agent_fields_introspection(self) -> List:
```
Comment (TimH98) [inline wandb/sdk/internal/internal_api.py:1533]: TODO: make sure this actually works if using an old version of wandb server

File: tox.ini [hunk 2]
```diff
@@ -154,7 +154,7 @@ deps=
 [testenv:func-{,grpc-,docs-,noml-,}py{37,38,39,310}]
 install_command=pip install --extra-index-url https://download.pytorch.org/whl/cpu {opts} {packages}
 deps =
-    yea-wandb=={env:YEA_WANDB_VERSION}
+    https://github.com/wandb/yea-wandb/archive/create-launch-agent-fields.zip
```
Comment (nickpenaranda) [inline tox.ini:157]: dev tracks? seems like this should not be committed to main

Comment (TimH98) [reply tox.ini:157]: Following along with the [functional testing guide](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#functional-testing), once this PR is approved and ready to ship, I'll merge my [yea-wandb change](https://github.com/wandb/yea-wandb/pull/118), change this to use the new yea-wandb version, then merge this change.

Comment (nickpenaranda) [reply tox.ini:157]: ah thanks, TIL

**Review by nickpenaranda — APPROVED** [review]

**Review by kptkin — APPROVED** [review]

**Pull request merged**